apps/builder/app/shared/nano-states/pages.ts

```diff
@@ -15,3 +20,19 @@ export const $selectedPage = computed(
     return findPageByIdOrPath(selectedPageId, pages);
   }
 );
+
+export const $existingRoutePaths = computed([$pages], (pages) => {
+  if (pages === undefined) {
+    return;
+  }
+
+  const paths = [];
```

Comment:
btw this should be a Map or a Set, Map has benefit for other use cases, where you get the page as a value